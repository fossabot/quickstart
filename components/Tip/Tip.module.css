.TipWrapper {
  margin-top: 40px;
  margin-bottom: 40px;
  font-size: 13px;
  line-height: 20px;
}

.TipWrapper svg {
  display: inline-block;
  width: 12px;
  height: 12px;
  margin-top: 3px;
  margin-right: 4px;
  vertical-align: top;
}

.TipWrapper svg path {
  fill: #1ea7fd;
}

.Tip {
  display: inline-block;
  margin-right: 10px;
  padding: 4px 12px;
  color: #66bf3c;
  font-weight: 700;
  font-size: 11px;
  line-height: 12px;
  vertical-align: top;
  background: #e7fdd8;
  border-radius: 1em;
}
