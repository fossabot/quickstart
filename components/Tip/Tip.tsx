import React, { ReactNode } from 'react'

import Styles from './Tip.module.css'

export interface TipProps {
  /**
   * Give the tip a label
   */
  label: string
  /**
   * The content to show inside the tip
   */
  children?: ReactNode
}

export const Tip: React.FC<TipProps> = ({ label, children }) => {
  return (
    <div className={Styles.TipWrapper}>
      <span className={Styles.Tip}>{label}</span> {children}
    </div>
  )
}
