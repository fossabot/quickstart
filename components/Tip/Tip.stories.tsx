import { Meta, Story } from '@storybook/react'
import React from 'react'

import { Tip, TipProps } from './Tip'

const config: Meta = {
  title: 'Components/Tip',
  component: Tip,
}

export default config

const Template: Story<TipProps> = (args) => <Tip {...args} />

export const Default = Template.bind({})
Default.args = {
  label: 'Tip',
}
Default.parameters = {
  jest: ['Tip.test.tsx'],
}

export const WithChildren = Template.bind({})
WithChildren.args = {
  label: 'Tip',
  children: <>Hello</>,
}
