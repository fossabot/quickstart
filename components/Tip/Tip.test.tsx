import React from 'react'

import { Tip, TipProps } from './Tip'

describe('Tip', () => {
  const props: TipProps = {
    label: 'Test Tip',
  }
  it('Should render without an error', () =>
    expect(() => <Tip {...props} />).not.toThrow())

  it('Should match the snapshot', () =>
    expect(<Tip {...props} />).toMatchSnapshot())
})
