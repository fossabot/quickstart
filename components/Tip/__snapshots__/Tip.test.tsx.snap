// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tip Should match the snapshot 1`] = `
<Tip
  label="Test Tip"
/>
`;
